src/ring_theory/polynomial/homogeneous.lean

```diff
@@ -291,6 +291,21 @@ begin
   exact λ h, (coeff_eq_zero_of_total_degree_lt h).symm
 end
 
+lemma homogeneous_component_homogeneous_polynomial (m n : ℕ)
+  (p : mv_polynomial σ R) (h : p ∈ homogeneous_submodule σ R n) :
+  homogeneous_component m p = if m = n then p else 0 :=
+begin
+  simp only [mem_homogeneous_submodule] at h,
+  ext x,
+  rw coeff_homogeneous_component, split_ifs, refl,
+  by_cases zero_coeff : coeff x p = 0, rw zero_coeff, simp only [coeff_zero],
+  exfalso, apply h_2, rw ←h_1,
```

Comment:
You are referring to the auto-generated names `h_2` and `h_1`. Please give them names in the tactic calls that introduced them. That makes the proof more robust, hence easier to maintain. It also improves readability.